junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/MethodBasedCondition.java

```diff
@@ -66,7 +66,7 @@ private Object evaluateCondition(Method method, ExtensionContext context) {
 		Preconditions.condition(areParametersSupported(method),
 			() -> format("method [%s] should take either an ExtensionContext or no parameters", method.getName()));
 
-		Object testInstance = context.getTestInstance().orElse(null);
+		Object testInstance = context.getRequiredTestInstance();
```

Comment:
@marcphilipp Am I wrong to assume that the `testInstance` will always be present in this case?
If so, is the presence of `@TestInstance(Lifecycle.PER_CLASS)` even required in the case of a non-static method?

Reply:
I think we're missing integration test for applying the condition on the class level in which case the test instance would not be present.